### PR TITLE
feat(pie-link): DSW-1057 add link states

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)
[Added] - link state styles

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-link/src/link.scss [hunk 1]
```diff
@@ -1,5 +1,17 @@
 @use '@justeattakeaway/pie-css/scss' as p;
 
+
+@mixin link-interactive-states($mode: 'default') {
+    &:hover,
```
Comment (kevinrodrigues) [inline packages/components/pie-link/src/link.scss:5]: Do we need focus states here?

Comment (raoufswe) [reply packages/components/pie-link/src/link.scss:5]: are you referring to the focus visible? if so, it's below

**Review by kevinrodrigues — APPROVED** [review]

File: apps/pie-storybook/stories/pie-link.stories.ts [hunk 2]
```diff
@@ -82,19 +92,27 @@ const linkStoryMeta: LinkStoryMeta = {
...
-            description: 'If `true`, makes the link text bold',
+            description: 'If `true`, makes the link text bold.',
             control: 'boolean',
             defaultValue: {
                 summary: false,
             },
         },
         isStandalone: {
-            description: 'If `true`, the link will be treated as a block box',
+            description: 'If `true`, the link will be treated as a block box.',
```
Comment (jamieomaguire) [inline apps/pie-storybook/stories/pie-link.stories.ts:102]: what does this mean @raoufswe ?

Comment (jamieomaguire) [reply apps/pie-storybook/stories/pie-link.stories.ts:102]: should it be 'block element'?

Comment (raoufswe) [reply apps/pie-storybook/stories/pie-link.stories.ts:102]: yea that would work too, block-level boxes are block containers/elements.

Comment (raoufswe) [reply apps/pie-storybook/stories/pie-link.stories.ts:102]: i will update it in my form-label PR 🥇

**Review by jamieomaguire — APPROVED** [review]

**Pull request merged**